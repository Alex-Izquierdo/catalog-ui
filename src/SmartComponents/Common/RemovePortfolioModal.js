import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import propTypes from 'prop-types';
import { Stack, StackItem, Button} from '@patternfly/react-core';
import { Main, PageHeader, PageHeaderTitle } from '@red-hat-insights/insights-frontend-components';
import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import { removePortfolio, fetchPortfolios } from '../../redux/Actions/PortfolioActions';

class RemovePortfolioModal extends Component {
  onSubmit = () => {
    this.props.removePortfolio(this.props.portfolio.id);
    this.props.closeModal();
  }

  onCancel = () => {
    this.props.addNotification({
      variant: 'warning',
      title: 'Removing portfolio',
      description: 'Removing portfolio was cancelled by the user.'
    });
    this.props.closeModal();
  }

  // Use pf4 components
  render() {
    return (
      <Main title={ 'Remove Portfolio' }>
        <Stack>
          <StackItem>
            <PageHeader>
              <PageHeaderTitle title= 'Remove Portfolio' />
            </PageHeader>
          </StackItem>
          <StackItem>
            <PageHeaderTitle>
              Remove Portfolio
            </PageHeaderTitle>
            <Button variant="primary" type="button" onClick={ this.onSubmit }>
              Remove
            </Button>
            <Button variant="primary" type="button" onClick={ this.onCancel }>
              Cancel
            </Button>
          </StackItem>
        </Stack>
      </Main>
    );
  }
}

const mapStateToProps = ({ portfolioReducer: { isLoading }}) => ({ isLoading });

const mapDispatchToProps = dispatch => bindActionCreators({
  addNotification,
  removePortfolio,
  fetchPortfolios
}, dispatch);

RemovePortfolioModal.propTypes = {
  isLoading: propTypes.bool,
  addNotification: propTypes.func,
  removePortfolio: propTypes.func,
  fetchPortfolios: propTypes.func,
  closeModal: propTypes.func,
  portfolio: propTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(RemovePortfolioModal);
