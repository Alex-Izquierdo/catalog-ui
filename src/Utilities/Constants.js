export const DEFAULT_PAGE_SIZE = 50;

const topoApiHost = process.env.TOPO_API_HOST || '';
const topoApiPort = process.env.TOPO_API_PORT || '443';
const topoBasePath = process.env.TOPO_BASE_PATH;
const topoApiVersion = process.env.TOPO_API_VERSION || '/api/v0.0';
export const SOURCES_API_BASE = `https://${topoApiHost}:${topoApiPort}${topoBasePath}${topoApiVersion.replace(/\/+$/, '')}`;

const apiHost = process.env.API_HOST || 'localhost';
const apiPort = process.env.API_PORT || '5000';
const basePath = process.env.BASE_PATH || '';
const apiVersion = process.env.API_VERSION || '/api/v0.0';
export const SERVICE_PORTAL_API_BASE = `https://${apiHost}:${apiPort}${basePath}${apiVersion.replace(/\/+$/, '')}`;
