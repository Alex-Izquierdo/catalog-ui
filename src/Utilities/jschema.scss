@import "../App.scss";
@import '../../node_modules/bootstrap/dist/css/bootstrap.css';


.rjsf {
  border: 10px;
  padding-top: 20px;
  padding-left: 20px;
  padding-right: 20px;
  max-width: 500px;

  font-size: small;
}
.form-group {
  margin-left: 15px;
  margin-bottom: 15px;
}


.hidden {
  display: none;
}



