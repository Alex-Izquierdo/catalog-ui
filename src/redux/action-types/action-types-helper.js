export const createAsyncActionTypes = (actionTypes, prefix) => actionTypes.reduce((acc, curr) => [
  ...acc,
  ...[ curr, `${curr}_PENDING`, `${curr}_FULFILLED`, `${curr}_REJECTED` ]
], []).reduce((acc, curr) => ({
  ...acc,
  [curr]: `${prefix}${curr}`
}), {});

