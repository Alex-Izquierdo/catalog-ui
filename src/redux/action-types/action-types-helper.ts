import { StringObject } from '../../types/common-types';

export const createAsyncActionTypes = (
  actionTypes: string[],
  prefix: string
): StringObject =>
  actionTypes
    .reduce<string[]>(
      (acc, curr) => [
        ...acc,
        ...[curr, `${curr}_PENDING`, `${curr}_FULFILLED`, `${curr}_REJECTED`]
      ],
      []
    )
    .reduce(
      (acc, curr) => ({
        ...acc,
        [curr]: `${prefix}${curr}`
      }),
      {}
    );
