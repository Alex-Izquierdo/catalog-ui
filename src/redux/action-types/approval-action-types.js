import { createAsyncActionTypes } from './action-types-helper';

const APPROVAL_PREFIX = '@@catalog/approval/';

const asyncActionTypes = [ 'FETCH_WORKFLOWS', 'LINK_WORKFLOW', 'UNLINK_WORKFLOW', 'RESOLVE_WORKFLOW' ];

export const ASYNC_ACTIONS = {
  ...createAsyncActionTypes(asyncActionTypes, APPROVAL_PREFIX)
};

