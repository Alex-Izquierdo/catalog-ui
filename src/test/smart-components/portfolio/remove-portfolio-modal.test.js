import React from 'react';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' ;
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware, ADD_NOTIFICATION } from '@red-hat-insights/insights-frontend-components/components/Notifications';

import RemovePortfolioModal from '../../../SmartComponents/Portfolio/remove-portfolio-modal';
import { CATALOG_API_BASE } from '../../../Utilities/Constants';
import { REMOVE_PORTFOLIO, FETCH_PORTFOLIOS } from '../../../redux/action-types';

describe('<RemovePortfolioModal />', () => {
  let initialProps;
  let initialState;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;

  const ComponentWrapper = ({ store, children }) => (
    <Provider store={ store }>
      <MemoryRouter initialEntries={ [ '/foo', '/foo/123', '/foo' ] } initialIndex={ 1 }>
        { children }
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      id: '123'
    };
    initialState = {
      portfolioReducer: {
        portfolios: [{
          id: '123'
        }]
      }
    };
    mockStore = configureStore(middlewares);
  });

  it('should render correctly', () => {
    const wrapper = shallow(<RemovePortfolioModal { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should call cancel action', () => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <ComponentWrapper store={ store }>
        <Route path="/foo/:id" render={ (...args) => <RemovePortfolioModal { ...args } { ...initialProps } /> } />
      </ComponentWrapper>
    );
    wrapper.find('button').first().simulate('click');
    expect(wrapper.find(MemoryRouter).children().props().history.location.pathname).toEqual('/foo');
  });

  it('should call remove action', (done) => {
    expect.assertions(3);
    const store = mockStore(initialState);

    apiClientMock.delete(`${CATALOG_API_BASE}/portfolios/123`, mockOnce((req, res) => {
      expect(req).toBeTruthy();
      return res.status(200);
    }));

    apiClientMock.get(`${CATALOG_API_BASE}/portfolios`, mockOnce((req, res) => {
      expect(req).toBeTruthy();
      return res.status(200).body({ data: []});
    }));

    const wrapper = mount(
      <ComponentWrapper store={ store }>
        <Route path="/foo/:id" render={ (...args) => <RemovePortfolioModal { ...args } { ...initialProps } /> } />
      </ComponentWrapper>
    );
    const expectedActions = [{
      type: `${REMOVE_PORTFOLIO}_PENDING`,
      meta: expect.any(Object)
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({ variant: 'success' })
    }), expect.objectContaining({
      type: `${REMOVE_PORTFOLIO}_FULFILLED`
    }), expect.objectContaining({
      type: `${FETCH_PORTFOLIOS}_PENDING`
    }), expect.objectContaining({
      type: `${FETCH_PORTFOLIOS}_FULFILLED`,
      payload: []
    }) ];

    wrapper.find('button').last().simulate('click');
    setImmediate(() => {
      expect(store.getActions()).toEqual(expectedActions);
      done();
    });
  });
});
