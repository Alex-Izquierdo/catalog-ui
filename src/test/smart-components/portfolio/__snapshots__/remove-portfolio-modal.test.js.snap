// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemovePortfolioModal /> should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
